### ISSUE TITLE
"no match for call" with uint64

### ISSUE THREAD

**Issue by Dapid** [serge-sans-paille/pythran#819]
The following script fails to compile with a large amount of compiler output. The same computation in uint32 or int64 compiles.

```
import numpy as np


def _compute_freqs(Z, n_cols, depth, q):
    s = q - 1
    expanded_cols = n_cols * s

    Pi = np.zeros(expanded_cols + 1, dtype=np.float64)
    Pij = np.zeros((expanded_cols + 1, expanded_cols + 1), dtype=np.float64) # Removing this variable and it works

    for i in range(n_cols):
        for k in range(depth):
            a = Z[k, i]
            Pi[i + a - 1] += (a != q)
    return Pi, Pij


# pythran export _compute_new_frequencies(uint64[:, :], int, int)
def _compute_new_frequencies(alignment, n_cols, depth):
    q = alignment.max()
    Pi_true, Pij_true = _compute_freqs(alignment, n_cols, depth, q)
    return Pi_true, Pij_true
 ```

Beginning of compiler error:
```
/tmp/tmp10ex_yop.cpp:60:152: error: no match for call to ‘({anonymous}::pythonic::numpy::functor::zeros) ({anonymous}::pythonic::types::array<long unsigned int, 2>, {anonymous}::pythonic::numpy::functor::float64)’
       typedef decltype(std::declval<typename std::remove_cv<typename std::remove_reference<decltype(pythonic::numpy::functor::zeros{})>::type>::type>()(std::declval<__type31>(), std::declval<__type15>())) __type33;
                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by serge-sans-paille** [comment]
Thanks for the restless reports :-)

This one was relatively easy to solve: `expanded_cols` is a uint64 wich was not  avalid shape type for pythran. I'll push a patch soon but meanwhile you can just use the following version:

```python
Pij = np.zeros((int(expanded_cols) + 1, int(expanded_cols) + 1), dtype=np.float64) # note the int cast
```

**Comment by Dapid** [comment]
No problem, anything to make it better! One of this days I will study the
developing tutorial, so I can contribute some code.

**Comment by serge-sans-paille** [comment]
Should be fided by #820

**Comment by serge-sans-paille** [comment]
Closed by #820
